pkg/middlewares/replicate/replicate.go

```diff
@@ -0,0 +1,72 @@
+package replicate
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/traefik/traefik/v2/pkg/log"
+	"github.com/traefik/traefik/v2/pkg/middlewares"
+)
+
+const (
+	typeName = "Replicate"
+)
+
+// replicate is a middleware used to send copies of requests and responses to arbitrary service
+type replicate struct {
+	next     http.Handler
+	name     string
+	producer Producer
+}
+
+// New creates a new http handler.
+func New(ctx context.Context, next http.Handler, producer Producer, name string) (http.Handler, error) {
+	log.FromContext(middlewares.GetLoggerCtx(ctx, name, typeName)).Debug("Creating middleware")
```

Comment:
Are you creating an instance here or am I mistaken? "Creating a middleware instance"?